core/src/main/java/hudson/PluginManager.java

```diff
@@ -219,6 +230,14 @@ public Api getApi() {
     }
 
     /**
+     * If non-null, the base directory for all exploded .hpi/.jpi plugins.
+     * @return the base directory for all exploded .hpi/.jpi plugins or {@code null} to leave this up to the strategy.
+     */
+    public File getWorkDir() {
```

Comment:
@CheckForNull?

but I wonder why given this is public that you don't just return the correct workDir (rootDir) and make this `@NonNull`?